// igdb-api-node/lib/perform-request.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _requestPromise = _interopRequireDefault(require("./request-promise"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = function _default(url, apiService) {
  var options = {
    headers: {
      Accept: 'application/json'
    },
    url: url
  };

  if (apiService.token) {
    options.headers.Authorization = "Bearer ".concat(apiService.token);
  }

  options.headers[apiService.keyHeader] = apiService.key;
  return (0, _requestPromise.default)(options).then(function (result) {
    return new Promise(function (resolve, reject) {
      if (result.response.statusCode !== 200) {
        reject(new Error("HTTP Status ".concat(result.response.statusCode, " - ").concat(options.url)));
        return;
      }

      try {
        resolve({
          body: JSON.parse(result.body),
          headers: result.response.headers,
          scrollCount: result.response.headers['x-count'] || result.response.headers['X-Count'],
          scrollUrl: result.response.headers['x-next-page'] || result.response.headers['X-Next-Page'],
          url: options.url
        });
      } catch (error) {
        reject(error);
      }
    });
  });
};

exports.default = _default;
